src/superphot_plus/supernova_class.py

```diff
@@ -0,0 +1,111 @@
+from enum import Enum
+
+
+class SupernovaClass(str, Enum):
+    """Classes of Supernovae"""
+
+    SUPERNOVA_IA = "SN Ia"
+    SUPERNOVA_IBC = "SN Ibc"
+    SUPERNOVA_II = "SN II"
+    SUPERNOVA_IIN = "SN IIn"
+    SUPERLUMINOUS_SUPERNOVA_I = "SLSN-I"
+    SUPERLUMINOUS_SUPERNOVA_II = "SLSN-II"
+
+    @classmethod
+    def get_type_maps(cls, allowed_types=["SN Ia", "SN II", "SN IIn", "SLSN-I", "SN Ibc"]):
+        """For some allowed supernova classes, create dictionaries
+        mapping an integer classification to the type string.
+
+        Parameters
+        ----------
+        cls : SupernovaClass
+            The SupernovaClass class.
+        allowed_types : list of str, optional
+            The list of supernova classes to be included in the mapping.
+            Defaults to ["SN Ia", "SN II", "SN IIn", "SLSN-I", "SN Ibc"].
+
+        Returns
+        -------
+        tuple of dict
+            Tuple containing the mapping of string labels to integer identifiers
+            and the mapping of integer identifiers to string labels, respectively.
+        """
+        allowed_types = [a_type.value if isinstance(a_type, cls) else a_type for a_type in allowed_types]
```

Comment:
Can you add a test for passing in either strings, enum values, or a mix?

Reply:
Sure, I'll do it in a second! :)